scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2929,7 +2951,14 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   }
 
   def simpleExprRest(t: Term, canApply: Boolean): Term = atPos(t, auto) {
-    if (canApply) newLineOptWhenFollowedBy[LeftBrace]
+    if (canApply) {
+      if (dialect.allowSignificantIndentation) {
+        newLineOptWhenFollowedByIndentationAnd[LeftBrace]
+        newLineOptWhenFollowedByIndentationAnd[LeftParen]
```

Comment:
you perform a ton of the same checks... i'd merge into one

Reply:
Good idea!